rebasehelper/utils.py

```diff
@@ -138,6 +142,83 @@ def cprint(cls, message, fg=None, bg=None, style=None):
         else:
             print(message)
 
+    @staticmethod
+    def color_is_light(rgb):
+        """Function to determine whether a color is dark or bright
+
+        Args:
+            rgb(str): rgb hexcode of the color, e.g 000000 for black
+
+        Return:
+            True or False depending on the color brightness
+
+        """
+        red, green, blue = int(rgb[:2], 16), int(rgb[2:4], 16), int(rgb[4:], 16)
+        brightness = 1 - (0.299 * red + 0.587 * green + 0.114 * blue) / 255
+        return brightness < 0.5
+
+    @staticmethod
+    def dark_or_bright_color_scheme():
+        """Get terminal background color and decide whether to use dark or bright color scheme
+
+        Return:
+            dark or bright depending on the background color
+
+        """
+        background_color = ConsoleHelper.exchange_control_sequence('\x1b]11;?\x07')
+        # use black color by default if the terminal doesn't support the control sequence
+        if background_color is None:
+            return 'dark'
+
+        # parse RGB from the return code, for example from 'rgb:0000/0000/0000' for black
+        background_color = background_color.replace('rgb:', '', 1)
+        rgb = ''.join([x[0:2] for x in background_color.split('/')])
```

Comment:
I don't get it. You are for some reason just ignoring half of 16-bit color components (which could also be 4-bit, 8-bit or 12-bit btw, see [`RGB Device specification`](https://tronche.com/gui/x/xlib/color/strings/rgb.html)), joining it together and then splitting it again in `color_is_light()` method. Why?

Reply:
You should write a function to parse `RGB Device specification` and return RGB tuple, without limiting precision of components.